lib/modules/showImages.js

```diff
@@ -583,22 +586,24 @@ export function toggleThingExpandos(thing, scrollOnExpando) {
 	}
 }
 
-export function mediaBrowseMode(oldThing, newThing) {
-	if (autoExpandActive) return false;
+function mediaBrowse(selected, unselected, options) {
+	if (!selected || !options.mediaBrowse || autoExpandActive) return;
 
-	const oldExpando = oldThing && oldThing.getEntryExpando();
-	const newExpando = newThing && newThing.getEntryExpando();
+	const oldExpando = unselected && unselected.getEntryExpando();
+	const newExpando = selected && selected.getEntryExpando();
 
 	if (oldExpando) {
 		mediaBrowseModeActive = oldExpando.expandWanted || oldExpando.open;
 	}
 
 	if (mediaBrowseModeActive) {
 		if (oldExpando) oldExpando.collapse();
-		if (newExpando) newExpando.expand();
-	}
 
-	return newExpando && mediaBrowseModeActive;
+		if (newExpando) {
+			newExpando.expand();
+			options.didAutoExpand = true;
```

Comment:
Does https://github.com/honestbleeps/Reddit-Enhancement-Suite/pull/3410/commits/ea08234b43bc55a26420520810af898b21244e08 look OK to you?